File: 1128/1128.c
int numberOfEquivalentDominoPairs(struct TreeNode* root) {
    if (!root) return 0;
    int l = numberOfEquivalentDominoPairs(root->left);
    int r = numberOfEquivalentDominoPairs(root->right);
    return 1 + (l > r ? l : r);
}

File: 1128/1128.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var numberOfEquivalentDominoPairs = function(root) {
    if (!root) return 0;
    return 1 + Math.max(numberOfEquivalentDominoPairs(root.left), numberOfEquivalentDominoPairs(root.right));
};